tensorflow/python/keras/engine/training_utils_v1_test.py

```diff
@@ -0,0 +1,91 @@
+      training_utils_v1.standardize_sample_or_class_weights(
+          {'invalid_key': 1.0}, ['output'], 'class_weight')
+
+  def test_invalid_float_class_weight_keys(self):
+    """Test that float keys in class_weight raise ValueError."""
+    with self.assertRaisesRegex(ValueError, "Invalid class_weight key.*0.5"):
+      training_utils_v1.standardize_sample_or_class_weights(
+          {0.5: 1.0}, ['output'], 'class_weight')
+
+  def test_valid_integer_class_weight_keys(self):
+    """Test that valid integer keys work correctly."""
+    result = training_utils_v1.standardize_sample_or_class_weights(
+        {0: 1.0}, ['output'], 'class_weight')
+    self.assertIsNotNone(result)
+
+  def test_valid_string_number_class_weight_keys(self):
+    """Test that valid string number keys work correctly."""
+    result = training_utils_v1.standardize_sample_or_class_weights(
+        {'0': 1.0}, ['output'], 'class_weight')
+    self.assertIsNotNone(result)
+
+  def test_valid_output_name_class_weight_keys(self):
+    """Test that valid output names work correctly."""
+    result = training_utils_v1.standardize_sample_or_class_weights(
+        {'output': 1.0}, ['output'], 'class_weight')
+    self.assertEqual(result, [1.0])
+
+  def test_sample_weight_not_affected(self):
+    """Test that sample_weight validation is unaffected."""
+    result = training_utils_v1.standardize_sample_or_class_weights(
+        {'output': [1.0, 2.0]}, ['output'], 'sample_weight')
+    self.assertEqual(result, [[1.0, 2.0]])
+
+  def test_multi_output_invalid_class_weight_keys(self):
+    """Test invalid keys in multi-output class_weight."""
+    class_weight = {
+        'output1': {'invalid_key': 1.0, 0: 2.0},
+        'output2': {0: 1.0, 1: 2.0}
+    }
+    with self.assertRaisesRegex(ValueError, 
+                                "Invalid class_weight key.*invalid_key.*"
+                                "output1"):
+      training_utils_v1.standardize_sample_or_class_weights(
+          class_weight, ['output1', 'output2'], 'class_weight')
+      
+  def test_data_adapter_valid_class_weight(self):
+    """Test valid integer and string keys in class_weight are processed 
+    correctly."""
+    fn = data_adapter._make_class_weight_map_fn({'0': 0.5, 1: 1.5})
+    self.assertTrue(callable(fn))  # Should return a function
+
+  def test_data_adapter_invalid_key_type(self):
+    """Test that float class_weight key raises ValueError."""
+    with self.assertRaisesRegex(ValueError, "Invalid class_weight key.*0.5"):
+      data_adapter._make_class_weight_map_fn({0.5: 1.0})
+
+  def test_data_adapter_non_contiguous_keys(self):
+    """Test that non-contiguous integer keys raise ValueError."""
+    with self.assertRaisesRegex(ValueError, 
+                                "Expected `class_weight` to be a dict.*"):
+      data_adapter._make_class_weight_map_fn({0: 0.3, 2: 0.7}) 
+
+
+if __name__ == '__main__':
+  tf.test.main()
```

Comment:
Please configure your editor to insert newline at the end of all files. I already gave a review feedback on this vein in this PR.